site/content/articles/monitoring-multiple-openshift-clusters.adoc

```diff
@@ -0,0 +1,74 @@
+
+It is not in the scope of Prometheus even to proper visualize those metrics, the UI is very simple and you would use Grafana for the graphing. 
+
+To keep things dead simple, there isn't any replication mechanism either.
+
+== The problems with prometheus
+
+This simplicity though comes with a cost: 
+
+*A*. Grafana can only query one Prometheus at a time which means that your monitoring server is a single point of failure: Having a single Prometheus instance per cluster is a single point of failure. 
+
+***
+Any configuration error or hardware failure could potentially result in the loss of important insights. Even a simple rollout could be a small disruption in the metric collection because a restart can be significantly longer than the scraping interval.
+***
+
+*B*. Storing on disc, on simple text files is not effective: reliability, availability, storage 
+space and speed can be an issue when you want to store a lot of data 
+
+== How do we fix them
+Let's try to fix the issues one at a time: 
+
+In order to fix *A* we could use one cool feature of the sofware that we call "Federation", this basically means that you could use an extra prometheus to query other isntances, this means that each of your cluster could run one prometheus, data would be queried by another instance installed outside of the cluster, in that way each cluster could be monitor on its own, with the extra possibility to have some sort of a supervisor. 
+
+The problem with the approach above would be that it will accentuate the problem *B*: *_you would need more storage._*
+
+=== The Thanos approach 
+
+one of the software who caught my attention lately is link:https://thanos.io/[Thanos]
+
+They're approach is pretty smart and consist on putting a central query layer amongst multiple Prometheus instance: they do it using a sidecar container that stores the data and serves them over Thanos’s canonical gRPC-based Store API, which allows selecting time series data by labels and time range. On top of this there is a querier component that queries them in a coordinated way using the same language as prometheus (PromQL). 
+
+To mitigate the storage problem each sidecar container can upload data into an object storage (S3). 
+
+The problem with that is the amount of storage needed, to improve this situation, Thanos uses a compactor that is mainly needed for downsampling. 
+
+== The Telegraf and InfluxDB approach
+
+The above seems to be one way to solve *A* but a some time ago I got an interesting problem from a customer: They do not only have multiple openshift clusters, all their existing monitoring system was relying on telegraf, they have good people in house taking care of it and they use it also to monitor the underlying infrastructure. Would have really been a problem to migrate all this work to something else. 
```

Comment:
Comma after 'solve A'.

Capitalise OpenShift

system should be plural (systems)

were relying on Vs was relying on

Full stop after telegraf

Last sentence should begin with 'It would have been..'